### DISCUSSION TITLE
Other ideas for smoother ender operations.

### DISCUSSION THREAD

**Discussion by Fun4Dez** [mriscoc/Ender3V2S1#761] [Ideas]
First of all, I love the firmware:
When changing filaments can you add a nozzle temp tune function for different filaments?
When you pause the print and find a flaw and are wanting to cancel it you can't without waiting for it to resume then scrambling to abort it, a cancel print button needs to be added on screens like the pause.
Change filament needs a cancel button in case it is selected by accident.
The same thing needs to be added to the filament runout sensor for the same reason of accidentally selecting or not ending up needing it.
It ought to have a hardware auto level selection button and to perform mesh build every so many prints.
The x-axis raises too high when the print is finished, be nice to set that along with the other firmware settings accessible with your firmware.
A hardware nozzle size selector similar to a Prusa that checks nozzle size written into .gcode.
Can you scroll the filename during the print and add in a time till filament change readout like a Prusa?
Another thing that Prusa does that is too handy not to have it on all printers is the holding down the encoder button for a couple seconds enables live z-axis movement.
Can you set up a Bowden Tube or Direct Extruder type Retract setting so that anyone with any ender 3 can run the same ender 3 print file regardless of the extruder type?
I wish it was as easy as typing in suggestions to enable any setting or option thru the firmware.
Fun4_Dez!
Anyone want to play the best video TAG game ever? Motocross Madness 2, I have everything to play.

**Comment by mriscoc** [comment]
Hi, thank you.

We have RCT6 program memory space restrictions, I'm prioritizing only changes that can have effect in printing experience and cannot be emulated by G-code config files.

For the retract thing I suggest using the "[firmware retract](https://github.com/mriscoc/Ender3V2S1/wiki/Firmware-Retraction)" feature, so the same G-code can be printed in Bowden or direct drive extruders without re-slicing the model.  

Scrolling the filename consumes RAM and CPU cycles while printing, the status line already has auto scroll capabilities and you can write the filename there using a plugin.

The printer doesn't know when a filament change is going to happen, maybe a slicer plugin is necessary to show that in the status line.

The final position after a print is finished is managed by the slicer end G-code script.  

In the Prepare/Filament Management menu there is a Pre-heat submenu over the filament change item.

You will able to disable filament runout sensor in the tune menu.

I agree about the cancel buttons for filament change and other functions.

**Comment by Fun4Dez** [comment]
Thanks, for being professional about your work, I wish others understood how important that is.
I work on medical lasers and every little item has documentation.
I didn't see anything on the Ender but a simple assembly manual, I've been hoping for something to come out.
Do or will you have a pdf for better words explaining what each setting and its options?
Thank you again.
WRN


________________________________

**Reply by mriscoc** [reply]
Main documentation can be found in Marlin site, for example, https://marlinfw.org/meta/features/
The Wiki also has additional information that complements Marlin: https://github.com/mriscoc/Ender3V2S1/wiki
There are lots of web pages with great documentation and guides as: 
https://teachingtechyt.github.io/calibration.html
https://ellis3dp.com/Print-Tuning-Guide/

**Comment by Fun4Dez** [comment]
I couldn't find your contact link thankfully outlook had it saved, I hope you get this.
In my 2 years of printing, You're the only one that seems to actually know what they are talking about.

My Ender has been printing perfectly since your last firmware release.
A few strange issues like motion settings not being saved but great firmware.
The other printer companies are going with the 32bit board and full color displays and playing catch up.

My questions are:

  1.  Why won't the speed, acceleration and jerk settings hold?
  2.  Shouldn't the steppers voltage on a dual z-axis be increased?
  3.  Any idea what they should be set to?
  4.  I have a 42-40 stepper motor on both my x and y axis, what voltage should I run these at?

Thanks,
WRN
805-728-5218

**Reply by mriscoc** [reply]
>   1.  Why won't the speed, acceleration and jerk settings hold?  

Check your slicer start script, that could be overwriting your settings

>  2.  Shouldn't the steppers voltage on a dual z-axis be increased?

Because the force is now distributed between two motors, it is generally not necessary to increase the v-ref

> 3 and 4

Here are some suggestions: https://support.th3dstudio.com/helpcenter/creality-v4-2-x-tmc-boards-recommended-vref/

**Comment by Fun4Dez** [comment]
Since our last communication my z-axis is still not moving up enough, still a 20mm cube still measures 20mm tall.
I increased the voltage but it still won't move the z-axis high enough for each layer.
Only the first layer prints correctly and it's perfect.
You do know why a mechanic asks another for their advice now don't you?
It means they know the answer they just don't want to hear it.
The last time I encountered this problem was with a used and corrupted 4.2.7 motherboard.

Questions:

  1.  Could this be a simple case of a corrupted firmware?
  2.  Is there enough voltage/energy to run 42-40 steppers on the x and y-axis, 42-34's for the dual z-axis, and a 42-38 for the extruder?
  3.  Should I try doubling the z-axis steppers voltage? Last time I did that it raised too high but that was with a used and corrupted 4.2.7 motherboard.

I have a spare new 4.2.7 motherboard, steppers, hot-ends and everything else that can break.
thanks
WRN

**Reply by mriscoc** [reply]
Hmm, that sounds like step loss due to driver overheating.

**Comment by Fun4Dez** [comment]
Last message I was having trouble with my ender 3v2 and suspected the z-axis driver may have gone bad.

I started printing some PETG, first time I've printed anything but PLA and TPU on my 3v2.
I was successful in printing an 8hr print but when I tried to print the same item again, I kept getting Power Outage Errors, there were no power outages anywhere in the area and the 110v light I have on the printer never went out or flickered.
The power outage resume function did not resume in the correct place, it skipped to the beginning of the next layer.

I had an extra brand new 4.2.7 motherboard and since I was having the z-axis issue, I installed it.
I have 42-40 steppers on both my x and y axis along with the 42-34 dual z-axis steppers and a 42-38 extruder stepper. I set the vrefs to 1.38 x and y, 1.35 on the extruder and 1.25 on the dual z steppers.
Calibrated the bed and hot-end and had it printing perfectly with PLA on a 18hr print.
I'm still getting the Power Outage Errors but only when printing PETG and only after an hour or more.
I can say that when the temp was at 240/85c it errored sooner but with the temp around 230/75c it took longer.

I'm betting the motherboard is correctly sensing a 24v voltage/current drop.
My power supply voltage is 24.1v disconnected from the motherboard and 24.09v with the motherboard connected and turned on, shouldn't it be closer to 25v?
I have 119VAC at the plug should I just adjust the PSU 24v output higher, if so what voltage.
I know the PSU can be vented better which I am going to do.

Thanks
WRN

**Reply by mriscoc** [reply]
Change the SD card to discard media problems. If there is a problem with the memory card, the printer can hang and reboot, if power outage recovery is enabled, a recovery popup will appear.

**Comment by Fun4Dez** [comment]
Not sure if this reply was meant for me.
I'm having the power outage errors.
I have a fairly new micro-sd card in mine.



End of line.........



-------- Original message --------
From: Miguel Risco-Castillo ***@***.***>
Date: 7/25/23 5:31 PM (GMT-08:00)
To: mriscoc/Ender3V2S1 ***@***.***>
Cc: Fun4_Dez ***@***.***>, Author ***@***.***>
Subject: Re: [mriscoc/Ender3V2S1] Other ideas for smoother ender operations. (Discussion #761)


Change the SD card to discard media problems.

**Comment by Fun4Dez** [comment]
I have your email saved but I am not sure if you will get this message.

I installed your latest firmware from September on my Ender 3v2.
I printed a single layer 120mm disc to fine tune the bottom layer.
I printed a 40mm calibration cube, the print went swimmingly.
I went to try a 20mm cube and my machine has gone haywire.

  1.  It won't home properly
  2.  Only the y-axis homes correctly.
  3.  The x-axis goes haywire to the right and slams against the stop.
  4.  Whichever homing choice is selected, immediately the z-axis lifts around a centimeter then stops.
  5.  I tried the firmware from May, and it won't auto-home properly either, it acts exactly the same.

I think the motherboard has gone bad again, this would be the 4th time, 3 x 4.2.7 versions and the original 4.2.2 board.
They have all done the same thing when they have failed, auto-homing gone haywire.
A motherboard would have to last an entire year to use any type of extended warrantees.
Only official Creality boards purchased from them are eligible for a warranty in the first year.
In fact, withing the US all electronic items by law have to have a One Year full replacement or full refundable.

It's a bit strange that it broke immediately after the new bios was installed, might the new version have caused it?

WR Nesbitt
805-728-5218

**Reply by mriscoc** [reply]
Hi the failure could be in your printer power supply.

**Comment by Fun4Dez** [comment]
I sent this quite a while ago, the power supply voltage was set at 24v exactly.

I bumped it up to 25.4v and the entire machine came to life.

I sent another message last night.
I did your latest bios update and my printer worked for about an hour and went crazy when homing.
I re-flashed back to the old bios and it does the same thing.

Any chance the new bios caused the board to fritz out?
I'm not trying to put blame you, figured you'd want to know.
I have good ventilation across the motherboard.

What do you recommend the setting the 24v power supply at?
25.4v isn't too high.
I'm waiting on another 4.2.7 motherboard.

We build and repair medical laser equipment, scientific and show lasers.
My boss the "Laser God" says no way can it function set at exactly 24v any load would drop it under 24v instantly and that's what it was doing.

No one online line knows anything, they say it's set to
.
If I must replace motherboards every 12 or so months so be it.
They cost under $40.
Thanks again.
WRN
805-728-5218

Sent by the Master Control Program.

End of line.....
________________________________

**Reply by mriscoc** [reply]
It's not a good idea to increase the power supply voltage. The control board has its own voltage regulators for the electronics, but the stepper drivers will be stressed.
